assets/Shaders/default.frag

```diff
@@ -15,36 +16,31 @@ uniform float uFogEnd;
 uniform vec3  uFogColor;
 uniform float uFogDensity;
 uniform bool uFogIsLinear;
-uniform int uMaterialFlags;
 
 void main(void)
 {
-	vec4 textureColor = vec4(oColor);
-	if(uIsTexture)
-	{
-		textureColor *= texture2D(uTexture, oUv);
-	}
+	vec4 finalColor = uIsTexture ? texture2D(uTexture, oUv) : vec4(oColor.rgb, 1.0);
```

Comment:
Minor:
You're assuming colored vertices disable texturing.

At the minute, I don't think there's anything with colored vertices in the wild (only format that supports them IIRC is the X format with either of the new parsers), but is this a correct assumption?
If I'm reading the DirectX specification correctly ( https://docs.microsoft.com/en-us/windows/win32/direct3d9/per-vertex-color-state ) then this overrides the material properties ambient etc. colors, and I would assume texturing still works in this case.

Reply:
Oops. I fixed it.

Reply:
wait a minute.
I need to fix it a little more.